src/services/pokemon.ts

```diff
@@ -17,20 +17,17 @@ type PokeApiPokemon = {
 };
 
 export const getPokemon = async () => {
-  let stats: Pokemon[] = [];
   const response = await fetch("https://pokeapi.co/api/v2/pokemon");
   const data: PokeApiResults = await response.json();
 
-  for (const result of data.results) {
-    const response2 = await fetch(result.url);
-    const data2: PokeApiPokemon = await response2.json();
-    const pokemonStats: Pokemon = {
-      key: data2.id,
-      name: data2.species.name,
-      url: data2.species.url,
-      front_image: data2.sprites.front_default,
-    };
-    stats.push(pokemonStats);
-  }
-  return stats;
+  return Promise.all(
+    data.results.map((result) => fetch(result.url).then((r) => r.json()))
+  ).then((results: PokeApiPokemon[]) =>
+    results.map((result) => ({
```

Comment:
This collects all of the results into a new array. Promise.all is returned as a large promise that resolves as this array of Pokemon.